pyro/infer/autoguide/guides.py

```diff
@@ -337,6 +377,9 @@ def _setup_prototype(self, *args, **kwargs):
         if self.latent_dim == 0:
             raise RuntimeError('{} found no latent variables; Use an empty guide instead'.format(type(self).__name__))
 
+        # Initialize guide params
+        self.loc = nn.Parameter(self._init_loc())
```

Comment:
`.loc` is an implementation detail of a few derived classes, but is not common to all. E.g. `AutoIAFNormal` has no `.loc`. Could you move this attribute down to derived classes: `Auto*Normal`, `AutoLaplaceApproximation`?